phys2bids/physio_obj.py

```diff
@@ -0,0 +1,403 @@
+
+    if len(var) < data_size:
+        is_valid(token, type(var))
+        var = var + [token] * (data_size - len(var))
+
+    return var
+
+
+class blueprint_input():
+    """
+    Main input object for phys2bids.
+    Contains the blueprint to be populated.
+    !!! Pay attention: there's rules on how to populate this object.
+    See below ("Attention") !!!
+
+    Input (Properties)
+    ------------------
+    timeseries : (ch, [tps]) list
+        List of numpy 1d arrays - one for channel, plus one for time.
+        Time channel has to be the first, trigger the second.
+        Contains all the timeseries recorded.
+        Supports different frequencies!
+    freq : (ch) list of floats
+        List of floats - one per channel.
+        Contains all the frequencies of the recorded channel.
+        Support different frequencies!
+    ch_name : (ch) list of strings
+        List of names of the channels - can be the header of the columns
+        in the output files.
+    units : (ch) list of strings
+        List of the units of the channels.
+
+    Properties
+    ----------
+    ch_amount: int
+        Number of channels (ch).
+
+    Optional properties
+    -------------------
+    num_timepoints_found: int
+        Amount of timepoints found in the automatic count,
+        *if* check_trigger_amount() is run
+
+    Methods
+    -------
+    rename_channels:
+        Changes the list "ch_name" in a controlled way.
+    return_index:
+        Returns the proper list entry of all the
+        properties of the object, given an index.
+    delete_at_index:
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+    check_trigger_amount:
+        Counts the amounts of triggers and corrects time offset
+        in "time" ndarray. Also adds property ch_amount.
+
+    Attention
+    ---------
+    The timeseries (and as a consequence, all the other properties)
+    should start with an entry for time and an entry for trigger.
+    Both should have the same length - hence same sampling. Meaning:
+    - timeseries[0] → ndarray representing time
+    - timeseries[1] → ndarray representing trigger
+    - timeseries[0].shape == timeseries[1].shape
+
+    As a consequence:
+    - freq[0] == freq[1]
+    - ch_name[0] = 'time'
+    - ch_name[1] = 'trigger'
+    - units[0] = 's'
+    - Actual number of channels (ANC) +1 <= ch_amount <= ANC +2
+    """
+    def __init__(self, timeseries, freq, ch_name, units):
+        self.timeseries = is_valid(timeseries, list, list_type=np.ndarray)
+        self.ch_amount = len(self.timeseries)
+        self.freq = has_size(is_valid(freq, list,
+                                      list_type=(int, float)),
+                             self.ch_amount, 0)
+        self.ch_name = has_size(ch_name, self.ch_amount, 'unknown')
+        self.units = has_size(units, self.ch_amount, '[]')
+
+    def rename_channels(cls, new_names, ch_trigger=None):
```

Comment:
This is a method for an instance of `blueprint_input`, not a class method, correct? If so then I think it would be great for this (and all subsequent `cls` params) to be revised to `self`! While this will still function as-written, it's very confusing to read!

```suggestion
    def rename_channels(self, new_names, ch_trigger=None):
```